docs/mkdocs.yml

```diff
@@ -1,4 +1,4 @@
-site_name: DSPy
+site_name: DSPy Documentation
```

Comment:
@okhat might be able to judge this better but I think we can keep this as DSPy? It's a mix of docs and landing page and everything in between imo